tests/libolm_migration.test.ts

```diff
@@ -0,0 +1,79 @@
+        testData.backupVersion = "3";
+        testData.backupRecoveryKey = "/FLbqTHzH1ihmQl3740Dm2aWgOzBng8HjYdGuCpuMLU=";
+        testData.crossSigningMasterKey = "oob99xn8lk3eXXERE9U/Zj6gFIsrmAgq3KGvE5Wr0r4=";
+        testData.crossSigningSelfSigningKey = "YH1IjbOdpOrIgYZRnQuTInLDV6iSzZ1bNs/UKvUOAII=";
+        testData.crossSigningUserSigningKey = "3SFl1AdH3egRKnP5OJZt9wJyamK/SEi8Pfw3dd0mPMo=";
+        await Migration.migrateBaseData(testData, pickleKey, store);
+
+        const session1 = new PickledSession();
+        session1.pickle =
+            "F2tPtegrPKM0c+8Gtw0yyPoQeJn7opKITs/SzFS0QH0uVT8aOTK52/N3p+ATQdWlN2BAsa8MGRXjPPUG+c5s9u/HeZKmpwSiqxgZ9DdbcFYuIy9wiOe4oV68Hu03Yr/vqb9LWPQMTDgSFi2z0u0OMoDCDPB417vztR6fzTE4rwE5HUHgWU1s/7tXcF26nMzeYHuhR8KmpAYgs2/Xt/hcSdsRsyjIVxg4II32gM7XhgYcmQBQewmKasChtmX4V3ihxW6zwib9VwcN+q7XAg01QJyQY4+KSh6YYDSC5j+0on/jhcrpIC4i95i4fFc2Wv5EAVBPB//6TsXsu0s49mkp/H0ZshSeuf/J8Ip9NWI09kl9NM6pNPlalVQQoimFF/FWOovJ8iGQmRpCMmTeJa5CpELZPGXNAPec/eSFqLnSTjyYBFHroaJu9Q";
+        session1.senderKey = "1QkuYT/03gzKvMDmKQi5slJvfXECt+ca3/Ue3Cj+Cms";
+        session1.lastUseTime = new Date(1703693124932);
+        await Migration.migrateOlmSessions([session1], pickleKey, store);
+
+        // now open an olm machine using the store, and check the data.
+        const olmMachine = await OlmMachine.initialize(
+            new UserId(TEST_USER_ID),
+            new DeviceId(TEST_DEVICE_ID),
+            "testMigration",
+            "testPass",
+        );
+        expect(olmMachine.identityKeys.curve25519.toBase64()).toEqual("LKv0bKbc0EC4h0jknbemv3QalEkeYvuNeUXVRgVVTTU");
+        expect(olmMachine.identityKeys.ed25519.toBase64()).toEqual("qK70DEqIXq7T+UU3v/al47Ab4JkMEBLpNrTBMbS5rrw");
+
+        const backupKeys: BackupKeys = await olmMachine.getBackupKeys();
+        expect(backupKeys.backupVersion).toEqual("3");
+        expect(backupKeys.decryptionKey?.toBase64()).toEqual("/FLbqTHzH1ihmQl3740Dm2aWgOzBng8HjYdGuCpuMLU");
+
+        // TODO: figure out a way to test cross-signing key import
```

Comment:
Is this something you wanted to address before landing this?

Reply:
Not really, tbh. It looks *tricky*.